src/langcheck/metrics/en/_openai.py

```diff
@@ -129,38 +139,72 @@ def get_score(
-            return None, None
-
-        if assessment not in self._assessment_to_score_mapping:
+        config_structured_assessments = {
+            "seed": 123,
+            "functions": functions,
+            "function_call": {
+                "name": self._function_name
+            },
+            "model": "gpt-3.5-turbo"
+        }
+        config_structured_assessments.update(self._openai_args or {})
+
+        responses = self._call_api(
+            prompts=fn_call_messages,
+            config=config_structured_assessments,
+        )
+        function_args = [
+            json.loads(response.choices[0].message.function_call.arguments)
+            if response else None for response in responses
+        ]
+        assessments = [
+            function_arg.get(self._argument_name) if function_arg else None
+            for function_arg in function_args
+        ]
+
+        # Check if any of the assessments are not recognized.
+        for assessment in assessments:
+            if (assessment is None) or (assessment
+                                        in self._assessment_to_score_mapping):
+                continue
             # By leveraging the function calling API, this should be pretty
-            # rare, but we're dealing with LLMs here so nothing is absolute!
+            # rare, but we're dealing with LLMs here so nothing is
+            # absolute!
             print(f'OpenAI returned an unrecognized assessment: "{assessment}"')
-            print(f'Prompt that triggered the failure is:\n{prompt}')
-            return None, None
 
-        return self._assessment_to_score_mapping[
-            assessment], unstructured_assessment
+        return [
+            self._assessment_to_score_mapping[assessment]
+            if assessment else None for assessment in assessments
+        ], unstructured_assessments
+
+    def _call_api(self, prompts: Iterable[str | None],
+                  config: dict[str, str]) -> list[Any]:
+        # Generates input dict for API call. This procedure is separated as a
+        # method because yapf fails when there are too much nests.
+        def _generate_model_input(prompt: str | None) -> dict[str, Any]:
+            return {"messages": [{"role": "user", "content": prompt}], **config}
+
+        # A helper function to call the API with exception filter for alignment
+        # of exception handling with the async version.
+        def _call_api_with_exception_filter(model_input: dict[str, Any]) -> Any:
+            if model_input is None:
+                return None
+            try:
+                return self._client.chat.completions.create(**model_input)
+            except Exception as e:
+                return e
+
+        model_inputs = map(_generate_model_input, prompts)
+        responses = [
+            _call_api_with_exception_filter(model_input)
+            for model_input in tqdm(model_inputs)
```

Comment:
For all the other places, `langcheck.utils.progess_bar.tqdm_wrapper` (I just found the typo here) is used to show the progress bar. Could you follow that here too for consistency?

If the reason why you used `tqdm` here is that the `tqdm_wrapper` exhausts the `map` object by getting the `total=` value inside, I think we should provide the `total=` value somehow from the outside. Otherwise we cannot show the progress bar, and that is very important for the user experience.

Reply:
Sure, I updated the code to convert the input into `list` and use  `langcheck.utils.progess_bar.tqdm_wrapper`.

Reply:
Hmm it actually does not work because this line "exhausts" all the contents of the iterator and actual `tqdm` gets an empty iterator.
https://github.com/Alnusjaponica/langcheck/blob/9eb2e49286d1228a11b95dbfaf56180a2b063310/src/langcheck/utils/progess_bar.py#L16

Let's think about a good way to avoid that